modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -563,29 +726,28 @@ struct Infer: public cv::detail::KernelTag {
         return result;
     }
 
-    static void run(IECompiled &iec, const IEUnit &uu, IECallContext &ctx) {
+    static void run(IECompiled                            &iec,
+                    const IEUnit                          &uu,
+                    std::shared_ptr<IECallContext>        &&ctx,
+                    cv::gimpl::GIslandExecutable::IOutput &out,
+                    cv::gimpl::ie::SyncPrim               &sync) {
```

Comment:
The idea of the context is avoiding so much arguments for a method.

Reply:
IMO, store all of them in context is unclear and ugly. What do you think ?

Reply:
Discussed locally, fixed.

Reply:
Discussed locally, fixed.